### ISSUE TITLE
[REVIEW]: Logodetect: One-shot detection of logos in image and video data

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3124]
**Submitting author:** @maxpumperla (<a href="http://orcid.org/0000-0002-7801-4184">Max Pumperla</a>)
**Repository:** https://github.com/Heldenkombinat/Logodetect
**Branch with paper.md** (empty if default branch): 
**Version:** v0.1
**Editor:** @arokem
**Reviewers:** @szkafander, @nsaunier, @arfon
**Archive:** 10.5281/zenodo.6483152

 

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/05c48a22dca08424659ae672c89e6921/status.svg)](https://joss.theoj.org/papers/05c48a22dca08424659ae672c89e6921)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/05c48a22dca08424659ae672c89e6921"><img src="https://joss.theoj.org/papers/05c48a22dca08424659ae672c89e6921/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/05c48a22dca08424659ae672c89e6921/status.svg)](https://joss.theoj.org/papers/05c48a22dca08424659ae672c89e6921)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/Heldenkombinat/Logodetect" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@nsaunier & @abhishekbajpayee, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @arokem know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/Heldenkombinat/Logodetect">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@maxpumperla) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [ ] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/Heldenkombinat/Logodetect">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@maxpumperla) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @nsaunier, @abhishekbajpayee it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1109/tpami.2016.2577031 is OK
- 10.1109/CVPR.2016.90 is OK
- 10.1016/j.patcog.2019.106965 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.41 s (152.0 files/s, 21408.5 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Python                          34            652            551           2558
Jupyter Notebook                 8              0           3278            652
JSON                             6              0              0            414
Markdown                         4            100              0            326
TeX                              1              7              0             68
YAML                             4              5              0             34
Bourne Shell                     3             14              7             32
Dockerfile                       1              8              0             13
INI                              1              3              0             11
-------------------------------------------------------------------------------
SUM:                            62            789           3836           4108
-------------------------------------------------------------------------------


Statistical information for the repository '7fb2c8c1f5d13cc151d9953c' was
gathered on 2021/03/21.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
Jorge                           37          1181            823           23.60
Jorge Davila-Chacon              2           459              5            5.46
Max Pumperla                    52          4481           1542           70.93

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
Jorge Davila-Chacon          24            5.2          6.4                4.17
Max Pumperla               3727           83.2          3.4                9.82
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03124/joss.03124/10.21105.joss.03124.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03124/joss.03124/10.21105.joss.03124.pdf) :page_facing_up: :point_left:

**Comment by arokem** [comment]
@maxpumperla : a couple of initial comments from me: 

- Could you please add more information to the reference "Siamese neural networks for one-shot image recognition."? That reference currently only has authors, year and title. Is there a journal or conference? DOI? 
- Is `mtorch` a dependency of the software? Doesn't seem so, but I just wanted to make sure.

**Comment by jorgedch** [comment]
Hi @arokem, thank you for the feedback.

> * Could you please add more information to the reference "Siamese neural networks for one-shot image recognition."? That reference currently only has authors, year and title. Is there a journal or conference? DOI?

Yes, this is done now. Also I moved the paper files to a folder called _paper/_.

> * Is `mtorch` a dependency of the software? Doesn't seem so, but I just wanted to make sure.

You are right, `mtorch` is not a dependency, we only adapted its structure to our training pipeline.

**Comment by jorgedch** [comment]
OK, just noticed that moving the files to _paper/_ breaks the build so I reverted the change.

**Comment by whedon** [comment]
:wave: @abhishekbajpayee, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @nsaunier, please update us on how your review is going (this is an automated reminder).

**Comment by kyleniemeyer** [comment]
Hello @arokem, @nsaunier, @abhishekbajpayee: I'm just checking in on the status of this submission. Do you have ETAs on when you might be able to submit your reviews?

**Comment by arokem** [comment]
@nsaunier and @abhishekbajpayee : have you had a chance to take a look? Thanks!

**Comment by arokem** [comment]
@nsaunier and @abhishekbajpayee : could you please give an update on the status of your reviews of this paper? It's been several months since you agreed to review this.

**Comment by abhishekbajpayee** [comment]
@arokem I am unable to edit the checkboxes above even after accepting the invite and being logged in. Can you advise on what I should do?

Until then, some early feedback to @jorgedch:
- Using `pip install` to directly install from pypi takes way too much time. Is there any way to make the install size smaller?
- Provided instructions don't work as-is. The url in the git clone command should ideally be the https one so that anyone is able to clone. The `cd` command right after it has a typo (lowercase l in logodetect).
- On install using `pip install -e` with a local clone, I run into a `No module named 'skbuild'` error.

FYI I am using `python3` and consequently `pip3`.

**Comment by arokem** [comment]
@openjournals/joss-eics : sorry to bother, but it seems that @abhishekbajpayee, one of the reviewers on this article, is unable to tick the boxes in the comment at the top of this page. Could you please help with that? Thanks!

**Comment by danielskatz** [comment]
Hi @arokem - what I'm going to do next is something any editor can do ...

**Comment by danielskatz** [comment]
@whedon re-invite @abhishekbajpayee as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@abhishekbajpayee please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by arfon** [comment]
:wave: @nsaunier and @abhishekbajpayee – just checking in here. Do you think you might be able to complete your initial reviews in the next week or so? This submission has been waiting on your reviews for quite some time now.

**Comment by maxpumperla** [comment]
@jorgedch any chance we can find the time to address the above findings by @abhishekbajpayee above? Might be good to sync on a round of Q&A together.

@arfon @arokem is this the normal/expected process for a paper submission? I'm just asking because this smaller logodetect paper is a "test run" for more high-level projects that I'm running, and that would fit your journal quite well. I know it's been a strange couple of years with Covid, but it'd be great to have some sort of process and reliability.

**Comment by arfon** [comment]
> @arfon @arokem is this the normal/expected process for a paper submission? I'm just asking because this smaller logodetect paper is a "test run" for more high-level projects that I'm running, and that would fit your journal quite well. I know it's been a strange couple of years with Covid, but it'd be great to have some sort of process and reliability.

This review is taking longer than we'd usually expect but it's not completely out of the ordinary. Over the last ~1300 papers we've published our average review time is something around 3 months (http://www.theoj.org/joss-analytics/joss-submission-analytics.html#pre-reviewreview-time-over-time).

**Comment by nsaunier** [comment]
I have left unanswered comments about the code for weeks now. I am not doing anything else till they are addressed. Thanks!---Nicolas Saunier, ing. DrSc / Ph.D.Professeur titulaire / Full ProfessorDépartement des génies civil, géologique et des mines (CGM) PolytechniqueMontréal CIRRELT, RRSR, CIRODD, IVADO & Chaire MOBILITÉhttp://nicolas.saunier.confins.netOn Aug. 7, 2021 02:48, Max Pumperla ***@***.***> wrote:
@jorgedch any chance we can find the time to address the above findings by @abhishekbajpayee above? Might be good to sync on a round of Q&A together.
@arfon @arokem is this the normal/expected process for a paper submission? I'm just asking because this smaller logodetect paper is a "test run" for more high-level projects that I'm running, and that would fit your journal quite well. I know it's been a strange couple of years with Covid, but it'd be great to have some sort of process and reliability.

**Comment by jorgedch** [comment]
My apologies for the confusion to everyone, normally I take only a couple of days to address feedback. In the case of the feedback from @abhishekbajpayee, his comments are related to the installation process of the python package, and assumed that Max was going to address them as he implemented the packaging and is more familiar with it. Can you address this in the following days  @maxpumperla? Otherwise, I can make time on Thursday. 

It would be very much appreciated if the feedback of the reviewers is done in parallel and not depending on the response to the feedback of other reviewers. Please let us know if you have any feedback on the algorithmic side of the submission @nsaunier, I can work on it in parallel to my co-author.

**Comment by maxpumperla** [comment]
@jorgedch sure, I can take a look! I'll let everyone know once we have a new version of the package online.

**Comment by jorgedch** [comment]
Dear reviewers,

After some updates, currently fresh installs on Mac and Ubuntu are running the installation process and the tests as indicated in the README file. There are some things to improve on our side, but are not blockers. For example, the installation order in pip seems to produce an error as a pip package is required to build the wheel of another one in the requirements, and the installation order seems to generate a conflict some times. We are looking into it. 

Also some errors are shown in the logs when running the server for the browser, but are not related to our package and make no problems for the user in the browser. Everything else is fixed now :-)

FYI:

- [Mac] Error shows first time running pip install -e ".[tests, dev]":
Failed building wheel for matplotlib

- [Mac] Second time running pip install -e ".[tests, dev]" it shows:
DEPRECATION: matplotlib was installed using the legacy 'setup.py install' method, because a wheel could not be built for it. A possible replacement is to fix the wheel build issue reported above. You can find discussion regarding this at https://github.com/pypa/pip/issues/8368.

- [Mac] Third time running pip install -e ".[tests, dev]" it works without problems. Possibly related to cython needed as a dependency to build matplotlib wheel, a classic problem with the installation order in pip, so it disappears after a few runs when it is installed. Maybe we can instruct people to install cython by hand before running setup.py?

- [Mac and Ubuntu] Error shows only first time running python app.py:
192.168.0.205 - - [13/Aug/2021 19:20:17] "GET /favicon.ico HTTP/1.1" 404 -

**Comment by maxpumperla** [comment]
@nsaunier @abhishekbajpayee we've addressed your concerns and published `logodetect` version `1.1.2`. If you'd be so kind to have a look, that'd be much appreciated.

@nsaunier if there are further problems, it would be great to please link the GitHub issues you might be creating in the `logodetect` repo _to this thread_. We're getting a lot of GitHub notifications and it can be intractable to understand who is who across repos. 

Also, thanks @arfon for the clarification. I just wanted to get a better picture to adjust expectations. I completely understand if things take their time.

**Comment by arfon** [comment]
FYI I just emailed @abhishekbajpayee to see when they might be able to complete their review by.

**Comment by jorgedch** [comment]
Super, many thanks @arfon!

**Comment by maxpumperla** [comment]
hey @arfon and @arokem, I was wondering if there are any updates on this issue?

**Comment by arokem** [comment]
Yeah - I agree. We should also remove this reviewer from the google spreadsheet.

**Comment by arokem** [comment]
@szkafander : have you had a chance to review the paper and software?

@nsaunier : where do things stand on your end? Have all your concerns been addressed? I see that you have not checked all of the boxes in your check list yet.

**Comment by szkafander** [comment]
@arokem I have not been able to install or build the package so far. I saw some discussion about it in previous comments and I've been hoping those issues are addressed. I am still having issues with installation. I will post the errors as issues and wait for a resolution.

**Comment by szkafander** [comment]
@arokem I figured that you wanted to move on with this review since it has been pending for a while. This is my experience with the code so far:

Installation does not work on Windows. I could not find any indication in the docs about the supported operating systems so I assumed that it should work on Windows. I was able to install on Linux. Did not try on Mac.

The package works as intended on demo data. I noticed glitches on user data. I opened an issue in the package repo.

Otherwise the code is well-documented and complies with JOSS standards. The ML is pretty cool. I am sure that the glitches can be fixed easily.

I went ahead and checked all boxes in my review but I would suggest that the authors fix the issues before you accept the paper.

**Comment by jorgedch** [comment]
Thank you for your feedback @szkafander.

[Installation on Windows]
Good catch, the package is intended to be used in Linux and MacOS, so we'll clarify this in the documentation and setup file.

[Detection accuracy]
Indeed, it is the case that the accuracy of the logo detections of the provided model is not very high. We describe in the README the following:

"additionally to sharing with you a couple of algorithms to **get started with one-shot logo-detection**, the aim of this project is to develop a flexible architecture to **facilitate the comparison of different algorithms** for one-shot object recognition." and "The models that we are including in the repo achieved a **reasonable performance after a few training epochs.** However, if you would like to **improve their performance** you can find pointers to various datasets in [6], which can be used in the training part of this project."

So the aim of the project is not to provide the user with a highly performant pre-trained model, but with a playground to allow a standardized comparison of one-shot algorithms. However, as we continue to refine the training of models for the detection of logos (and other classes of objects, such as traffic signs, faces, etc.), we are very happy to share those improved models later on with the community.

Thank you once again for your feedback @szkafander. I'll update the repo with the changes that you point out about the OS 👍

**Comment by arokem** [comment]
Hello @nsaunier : have you had a chance to take another look at this? I see that several check-boxes are unchecked in your list. Could you provide some more comments, that the authors can address these issues? Thanks!

@jorgedch : have you had a chance to address the issues raised by @szkafander?

**Comment by jorgedch** [comment]
Hi @arokem. Yes, I made the changes the same day of my previous reply. The setup file now specifies that the intended OS is Linux only: `Operating System :: POSIX :: Linux`. Concerning the comment about the accuracy, it is also clarified why the provided model is not performing accurately with some images. Last time we heard from one of the reviewers was almost half a year ago. What is the journal policy for waiting times?

**Comment by arokem** [comment]
I am so sorry that this review is so slow. Unfortunately, @nsaunier is no longer responsive. I also tried to email him to his work email. It is understandable that folks have other things on their mind at this time. 

I am going to try to find another reviewer, with the hope that can expedite the process and that we can come to some conclusion here.

**Comment by arokem** [comment]
@xiaowuc2 : in the past, you agreed to take on this JOSS review. Would you still be willing to do so? It would be a great help.

**Comment by arokem** [comment]
@conradsnicta @vchuravy @abuccts: would any of you be willing to review this submission for JOSS? We carry out our checklist-driven reviews here in GitHub issues and follow these guidelines: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html

The submission under review here is [Logodetect: One-shot detection of logos in image and video data, a Python package for detecting logos in images and videos](https://github.com/Heldenkombinat/Logodetect).

**Comment by conradsnicta** [comment]
@arokem Apologies - I'm snowed under with work

**Comment by arfon** [comment]
Just leaving a quick note here I'm looking for another reviewer to wrap up this submission.

**Comment by arfon** [comment]
@editorialbot add @arfon as reviewer

**Comment by arfon** [comment]
@editorialbot generate my checklist

**Comment by arfon** [comment]
@editorialbot generate pdf

**Comment by arfon** [comment]
@maxpumperla @jorgedch – I'm going to do the second review here myself. If/as I find issues, I'll ask them as I go:

>  State of the field: Do the authors describe how this software compares to other commonly-used packages?

The paper doesn't really address this instead saying _'there is plenty of literature on the use of deep-learning for detecting logos'_. Would it be possible to add a couple of sentences to your paper saying something about other implementations/approaches compare to Logodetect?

**Comment by arfon** [comment]
Installing from source on a fresh GitHub Codespaces (after running `pip install -r requirements.txt`), I get the following error when initially attempting to start the `logodetect` CLI tool (as per docs [here](https://github.com/Heldenkombinat/Logodetect#usage)):

```
ImportError: libGL.so.1: cannot open shared object file: No such file or directory
```

I fixed this with `apt-get install libgl1`. Should `libgl1` included as part of the [`opencv` requirements](https://github.com/Heldenkombinat/Logodetect/blob/master/requirements.txt#L10)? If so, it doesn't seem to be 🤷 ?

**Comment by arfon** [comment]
Running the test examples, the image-based detector seems to do a worse job than the video detector. Is this to be expected?

![test_image_small_output](https://user-images.githubusercontent.com/4483/161925429-bb4376b3-4e53-4978-9d29-c2de252b7e18.png)

https://user-images.githubusercontent.com/4483/161925438-3fc36e05-6e93-4697-b60a-0f20ec6e11a9.mp4

**Comment by arfon** [comment]
OK, so in summary, I think the following issues need to be addressed for me to sign off on this review:

- Installation instructions: Resolving the potential issue around `libGL` dependency [[link](https://github.com/openjournals/joss-reviews/issues/3124#issuecomment-1089949375)].
- Addressing feedback on 'State of the field' in the paper [[link](https://github.com/openjournals/joss-reviews/issues/3124#issuecomment-1089925666)]
- Missing community guidelines [[link](https://github.com/openjournals/joss-reviews/issues/3124#issuecomment-1089969176)]

**Comment by jorgedch** [comment]
Thank you for the kind effort on your side @arfon, very much appreciated. I'll address your feedback this weekend. Cheers, Jorge

**Comment by jorgedch** [comment]
Hi @arfon, 

Thank you once again for your feedback, we have addressed it as follows:

> OK, so in summary, I think the following issues need to be addressed for me to sign off on this review:
> 
> * Installation instructions: Resolving the potential issue around `libGL` dependency [[link](https://github.com/openjournals/joss-reviews/issues/3124#issuecomment-1089949375)].

Done. We added the clarification that the library has a dependency on libgl1 and the command to install it. 

> * Addressing feedback on 'State of the field' in the paper [[link](https://github.com/openjournals/joss-reviews/issues/3124#issuecomment-1089925666)]

We have clarified the distinction of our library as follows:

"we also provide a modified ResNet [2] for object-recognition that directly takes the exemplars and the detections from the first stage and predicts if both belong to the same sub-class. Similarly to [3], this network infers a distance metric after being trained with examples of different sub-classes, **but instead of sharing the same weights and processing each input in a separate pass as in [4], it concatenates both inputs and processes them in one pass**. This concept follows more closely the architecture proposed by [5], where the assumption is that the exemplars often have more high-frequency components than the detections, and therefore the model can increase its accuracy by learning a separate set of weights for each input. **However, our proposed architecture splits the detection and classification tasks in two separate stages, which allows the use, and comparison, of different classifiers for the second stage.**"

> * Clarifying if the test behaviour with images and videos is expected [link](https://github.com/openjournals/joss-reviews/issues/3124#issuecomment-1089962647)

Yes, this behavior is expected. The baseline model uses a distance threshold to decide whether a detected logo belongs to one of the brands that the user is searching for. For this reason, in different video frames the classifier may behave differently between images that are fairly similar. However, the baseline model is the simplest approach and the reference model, as well as other models that the user wants to test, improve over the baseline.

> * Missing community guidelines [[link](https://github.com/openjournals/joss-reviews/issues/3124#issuecomment-1089969176)]

We have added now documentation for support, issues and contributions.

I believe this addresses the feedback points. Looking forward to the next steps :) All the best!

**Comment by arfon** [comment]
OK thanks @jorgedch! I've found one minor issue here that could do with fixing: https://github.com/Heldenkombinat/Logodetect/pull/40

At this point could you make a new release of this software that includes the changes that have resulted from this review. Then, please make an archive of the software in Zenodo/figshare/other service and update this thread with the DOI of the archive? For the Zenodo/figshare archive, please make sure that:

- The title of the archive is the same as the JOSS paper title
- That the authors of the archive are the same as the JOSS paper authors

I can then move forward with accepting the submission.

**Comment by arfon** [comment]
@editorialbot set 10.5281/zenodo.6483152 as archive

**Comment by arfon** [comment]
@editorialbot recommend-accept

**Comment by arfon** [comment]
@editorialbot accept

**Comment by arfon** [comment]
@szkafander, @nsaunier – many thanks for your reviews here and to @arokem for editing this submission! JOSS relies upon the volunteer effort of people like you and we simply wouldn't be able to do this without you ✨ 

@maxpumperla – your paper is now accepted and published in JOSS :zap::rocket::boom:. Sorry it took a while longer than expected 😅
